File: backend/backend-challenge/src/main/java/br/com/creditas/challenge/model/Product.java
package br.com.creditas.challenge.model;

import java.math.BigDecimal;

import br.com.creditas.challenge.model.enums.ProductType;

public class Product {

	private ProductType type;
	private BigDecimal price;
	
	public ProductType getType() {
		return type;
	}
	public void setType(ProductType type) {
		this.type = type;
	}
	public BigDecimal getPrice() {
		return price;
	}
	public void setPrice(BigDecimal price) {
		this.price = price;
	}
	
}

File: backend/src/main/kotlin/com/creditas/challenge/model/Product.kt
package com.creditas.challenge.model

import java.math.BigDecimal
import java.math.RoundingMode

data class Product(val name: String, val type: ProductType, private val _price: Double) {

    init {
        require(name.isNotBlank()) { "Product name cannot be blank" }
        require(_price > 0.0) { "Produce price must be > 0" }
    }

    val price: BigDecimal = _price
        .toBigDecimal()
        .setScale(2, RoundingMode.HALF_UP)
}

enum class ProductType {
    PHYSICAL,
    PHYSICAL_TAX_FREE,
    DIGITAL,
    SUBSCRIPTION
}
